src/weatherSource/WeatherSource.js

```diff
@@ -0,0 +1,129 @@
+import {
+    relative
+} from "path";
+
+const API_KEY = '953349aaa2569f9cd4821f8c2ffda23a'
+
+class WeatherAPI {
+
+    static _throw(text) {
+        throw text;
+    }
+
+    static _verify(params) {
+        let constructUrl;
+
+        if (params) {
+            params.latitude && params.longitude ? constructUrl = `lat=${params.latitude}&lon=${params.longitude}` :
+                params.city && params.country ? constructUrl = `q=${params.city},${params.country}` :
+                params.city ? constructUrl = `q=${params.city}` : this._throw('Wrong parameters')
+
+        } else {
+            throw 'No parameters were given'
+        }
+        return constructUrl;
+    }
+    // zwraca obiekt 1pogodowy
+    // w parametrach należy podać dwie wartości string:
+    // pierwszą 'miasto' i drugą 'państwo'
+    static async getWeatherByPlace(city, country) {
+
+        let dataUnparsed;
+
+        country
+            ?
+            dataUnparsed = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city},${country}&APPID=${API_KEY}&units=metric`) :
+            dataUnparsed = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city}&APPID=${API_KEY}&units=metric`)
+
+        if (dataUnparsed.ok) {
+            let data = await dataUnparsed.json();
+            // wyświetlenie całej zawartości odpowiedzi 
+            // console.log(data);
+
+            const dataModel = {
+                main: data.main,
+                clouds: data.clouds,
+                weather: data.weather,
+                wind: data.wind,
+                place: data.name,
+                coords: data.coord
+            }
+
+            return dataModel
+        }
+
+        throw dataUnparsed;
+    }
+
+    // zwraca obiekt pogodowy
+    // W parametrach należy przekazać obiekt z polami:
+    // latitude(szerokośc geo.) i longitude(długość geo.) 
+    static async getWeatherByCoordinates(coords) {
+
+        let dataUnparsed = await fetch(`http://api.openweathermap.org/data/2.5/weather?lat=${coords.latitude}&lon=${coords.longitude}&APPID=${API_KEY}`)
+        if (dataUnparsed.ok) {
+            let data = await dataUnparsed.json();
+            // console.log(data);
+
+            const dataModel = {
```

Comment:
To trzeba wywalić do osobnej metody prywatnej - popatrz, że w każdym wariancie pojawia się to samo. Co więcej, kiedy w końcu powstanie nam nasza wewnętrzna klasa Weather, to trzeba będzie to modyfikować. A jeśli klasa Weather się zmieni? To trzeba będzie modyfikować JESZCZE RAZ.
Ten kod zdecydowanie powinien zostać wyciągnięty do osobnej metody, żeby podobne zmiany dokonywane były tylko w jednym miejscu.

Reply:
Nie wiem, o Co konkretnie ci chodzi (żeby wywalić), bo zaznaczyłeś znaczą cześć kodu(koniec jednej metody, całą drugą i początek trzeciej). 

Jeżeli chodzi o walidacje parametrów. Jestem prawie pewny, że wyrzucę metody pobierające pogode `...ByPlace` oraz `...ByCoordinates`, a zostanę przy samym `..ByObject`. Wtedy zastosowanie mętody `_verify`, która już jest w kodzie, będzie adekwatniejsze. Mogę oczywiście je zostawić.

Jeżeli chodzi o konstrukcje obiektu `dataModel`, to już mam napisaną metodę co odchudzi obiekt `data`.